### PR TITLE
Add support in API for create, modify, delete mudlogs

### LINKED ISSUE #141: User story:  MUDLOG Object - API
As a Developer, I want the WE API to support the MUDLOG object so that I can use the WE API to perform operations on this object

### Acceptance criteria ###
- Ability to utilize the WE API to perform CRUD operations on the MudLog Object

Also, here are a few points that need to be addressed:
- [ ] Constraint 1
- [ ] Constraint 2

### Resources ###
-  Resource A
-  Resource B

### PR DESCRIPTION
**Pull request by haralde**
# Request Template Witsml Explorer
Thank you for contributing to WITSML Explorer!
Before submitting this PR, please fill in the following template.

## Fixes
This pull request fixes # (issue)

## Description
_Please include a summary of the change and which issue is fixed. List any dependencies that are required for this change._

API support for create, modify and delete mudlogs

## Type of change
_List the types of change. Remove from the list any points which are not necessary_

* New feature (non-breaking change which adds functionality)

## Impacted Areas in Application
_List general components of the application that this PR will affect:_

*

## Checklist:
_Please tick all the boxes or remove the ones that aren't needed and explain why_

*Communication*
* [ ] PR is related to an issue
* [ ] I have made corresponding changes to the documentation

*Code quality*
* [x] Code follows the style guidelines
* [x] I have self-reviewed my code
* [x] No new warnings are generated

*Test coverage*
* [x] Existing tests pass
* [ ] New code is covered by passing tests

## Further comments
_If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc..._

### CODE REVIEW

**Comment by steinsiv** [comment]
API endpoints 

```csharp
Routes.cs:            Get("/api/wells/{wellUid}/wellbores/{wellboreUid}/mudlogs", GetMudLogsForWellbore);
Routes.cs:            Get("/api/wells/{wellUid}/wellbores/{wellboreUid}/mudlogs/{mudlogUid}", GetMudLog);
```

was tested locally and everything worked fine at my end 👍 

I had to merge main into this branch and make some small changes in the following files to make it run.
This due to changes merged in main after this PR was published:

```csharp
/Src/WitsmlExplorer.Api/Services/MudLogService.cs
/Src/WitsmlExplorer.Api/Workers/CreateMudLogWorker.cs
/Src/WitsmlExplorer.Api/Workers/DeleteMudlogWorker.cs
```
Changes was in the use of `OptionsIn` generally and the new use of extending `IWorker<CreateMudlog>` in `CreateMudLogWorker.cs` 

I did not test the jobs locally

File: Src/WitsmlExplorer.Api/Models/MudLog.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+{
+    public class MudLog
+    {
+        public string WellUid { get; set; }
+        public string WellboreUid { get; set; }
+        public string Uid { get; set; }
+        public string WellName { get; set; }
+        public string WellboreName { get; set; }
+        public string Name { get; set; }
+        public string ObjectGrowing { get; set; }
```
Comment (lindsve) [inline Src/WitsmlExplorer.Api/Models/MudLog.cs]: Perhaps this could be boolean instead?

**Review by lindsve — APPROVED** [review]
LGTM, just a small comment about property type in the MudLog model class

- [x] Read issue and PR
- [ ] Pulled branch and manually tested
- [ ] Verify that PR resolves issue
- [x] Reviewed the code

**Comment by marmid74** [comment]
@haralde, please review comments from @steinsiv and @lindsve. 
Would be nice to get this one merged, with those suggested updates in place :-)

**Review by steinsiv — APPROVED** [review]
Tests run locally with recently merged main and docker images built and tested 👍

**Pull request merged**